src/manifest.c

```diff
@@ -323,6 +322,20 @@ static struct manifest *manifest_from_file(int version, char *component, bool he
 		c = c2;
 
 		file->filename = strdup_or_die(c);
+		/* If the file is an iterative manifest, copy the from_version and obtain the
+		 * bundle name based on the file name */
+		if (file->is_iterative == 1) {
+			char *ptr;
+			ptr = strrchr(file->filename, '.');
+			if (ptr != NULL) {
+				/* Keep only the bundle name from the file name, the filename should be
+				 * something similar to bundle-name.I.20 */
+				file->bundlename = strndup(file->filename, ptr - file->filename - 2);
```

Comment:
I forgot to comment on that. Can you create an strndup_or_die() function so we can handle strndup errors?

Reply:
sure thing